tests/integration/test_charm.py

```diff
@@ -61,3 +62,41 @@ async def test_with_ingress(
     )
     assert response.status_code == 200
     assert "Welcome to the Matrix" in response.text
+
+
+async def test_server_name_changed(model: Model, another_synapse_app: Application):
+    """
+    arrange: build and deploy the Synapse charm.
+    act: change server_name via juju config.
+    assert: the Synapse application should prevent the change.
+    """
+    unit = model.applications[another_synapse_app.name].units[0]
+    # Status string defined in Juju
+    # https://github.com/juju/juju/blob/2.9/core/status/status.go#L150
+    assert unit.workload_status == "blocked"
+    assert "server_name modification is not allowed" in unit.workload_status_message
+
+
+async def test_reset_instance_action(
+    model: Model, another_synapse_app: Application, another_server_name: str
+):
+    """
+    arrange: build and deploy the Synapse charm.
+    act: change server_name via juju config.
```

Comment:
I don't think the arrange and act are properly set
It should be something like
`arrange: a deployed Synapse charm in a blocked state due to a server_name change`
`act: call the reset_instance action

Reply:
Oops, you're right, thanks. Changed.